scrypto/src/macros.rs

```diff
@@ -17,7 +17,16 @@ macro_rules! error {
 #[cfg(not(feature = "log-error"))]
 #[macro_export]
 macro_rules! error {
-    ($($args: expr),+) => {{}};
+    ($($args: expr),+) => {{
+        // The operation below does nothing at runtime but consumes the expressions,
+        // allowing tools (e.g., rust-analyzer) to parse them correctly to identify
+        // syntactical errors or provide hints and completions.
+        // Using `false` ensures the expression is optimized out in case of expressions
+        // that mutate data or have side effects.
+        if false {
```

Comment:
I'm surprised this doesn't give an unreachable code lint. I wouldn't be surprised if such a lint were added some time. Perhaps we should make it this:
```rust
#[allow(unreachable_code)]
if false {
    let _ = ::scrypto::prelude::sbor::rust::format!($($args),+);
}
```

Reply:
Yeah, that is pretty surprising :) Done